src/Uno.UWP/UI/Input/ManipulationDelta.cs

```diff
@@ -12,5 +25,24 @@ public partial struct ManipulationDelta
 		/// <inheritdoc />
 		public override string ToString()
 			=> $"x:{Translation.X:N0};y:{Translation.Y:N0};θ:{Rotation:F2};s:{Scale:F2};e:{Expansion:F2}";
+
+		internal ManipulationDelta Add(ManipulationDelta right) => Add(this, right);
+		internal static ManipulationDelta Add(ManipulationDelta left, ManipulationDelta right)
+			=> new ManipulationDelta
+			{
+				Translation = new Point(
+					left.Translation.X + right.Translation.X,
+					left.Translation.Y + right.Translation.Y),
+				Rotation = left.Rotation + right.Rotation,
+				Scale = left.Scale * right.Scale,
+				Expansion = left.Expansion + right.Expansion
+			};
+
+		// Note: We should apply a velocity factor to thresholds to determine if isSignificant
+		internal bool IsSignificant(GestureRecognizer.Manipulation.Thresholds thresholds)
+			=> Math.Abs(Translation.X) >= thresholds.TranslateX
+			|| Math.Abs(Translation.Y) >= thresholds.TranslateY
+			|| Rotation >= thresholds.Rotate // We used the ToDegreeNormalized, no need to check for negative angles
```

Comment:
Threshold should be singular ? (same at other places)

Reply:
It's a bunch of multiple thresholds, I would keep it plural ... but I'm not an English expert 😁